challenge/dns01/nameserver.go

```diff
@@ -67,6 +79,21 @@ func AddRecursiveNameservers(nameservers []string) ChallengeOption {
 	}
 }
 
+// SetNetworkStack defines the IP stack that will be used for DNS queries.
+// Accepts "both", "ipv4", or "ipv6".
+func SetNetworkStack(network string) {
+	switch network {
+	case "ipv4":
+		currentNetworkStack = ipv4Only
+
+	case "ipv6":
+		currentNetworkStack = ipv6Only
+
+	default:
+		currentNetworkStack = defaultNetworkStack
+	}
+}
```

Comment:
Simplify the function to:

```suggestion
// SetNetworkStack defines the IP stack that will be used for DNS queries.
func SetNetworkStack(network networkStack) {
	currentNetworkStack = network
}
```